docs/cloud/tutorials/integrating-external-services.md

```diff
@@ -0,0 +1,114 @@
+# Integrating External Services
+
+Kubernetes, and by extension OpenShift OKd, gives a lot of flexibility regarding network use cases. One of the uses cases is the one that allows to use external services, like databases, transparently in a OpenShift project. Other of the use cases would be to to have a network proxy between OpenShift installations. This could be used as a temporal to ease migration periods.
+
+![Proxy between clusters](../img/proxy.drawio.svg)
+
+In the example above we are redirecting traffic from Rahti 1 to Rahti 2. This is obtained by creating two routes, a service and an Endpoint.
+
+## Procedure
+
+1. First deploy an application in Rahti 2. Any application is suitable for this test.
+
+1. [Install](../../rahti2/usage/cli/#how-to-install-the-oc-tool) and [login with OC](../../rahti2/usage/cli/#how-to-login-with-oc) in Rahti 2.
+
+1. Create a special `Route` in Rahti 2:
+
+    ```yaml
+    echo 'apiVersion: route.openshift.io/v1
+    kind: Route
+    metadata:
+      name: test-route
+    spec:
+      host: <test>.rahtiapp.fi
+      port:
+        targetPort: http
+      to:
+        kind: Service
+        name: <Service-name>
+        weight: 100
+      wildcardPolicy: None
+    status: {}' | oc create -f -'
+    ```
+    You can replace `<test>` by any available URL in Rahti 1. The `<service-name>` must be the one corresponding to the application you deployed. Double check which ports is `<Service-name>` exporting and adapt the Route in accordance to that.
+
+    You can test the setup so far:
+
+    ```sh
+    curl <test>.rahtiapp.fi -vL --resolve <test>.rahtiapp.fi:80:195.148.21.61
+    ```
+    The command above uses `--resolve` to chage the ip associated to a DNS that corresponds to Rahti 2.
+
+1. [Install](../../rahti/usage/cli/#how-to-install-the-oc-tool) and [login with OC](../../rahti/usage/cli/#how-to-login-with-oc) in Rahti 1.
+
+1. Create a project in Rahti 1, or use an existing one.
+
+1. Then create an `EndPoint` in Rahti 1:
+
+    ```yaml
+    echo 'kind: "Endpoints"
+    apiVersion: "v1"
+    metadata:
+      name: "proxy-service" 
+    subsets: 
+      -
+        addresses:
+          -
+            ip: "195.148.21.61"
+        ports:
+          -
+            port: 80
+            name: "http"' | oc create -f -
+    ```
+    The IP in the example, is the one behind `router-default.apps.2.rahti.csc.fi`. 
+
+1. Then a `Service`, also in Rahti 1:
+
+    ```yaml
+    echo 'kind: "Service"
+    apiVersion: "v1"
+    metadata:
+      name: "proxy-service"
+    spec:
+      ports:
+        -
+          name: "http"
+          protocol: "TCP"
+          port: 80
+          targetPort: 80 
+          nodePort: 0
+    selector: {}' | oc create -f -
+    ```
+    The name of the `Service` and the `EndPoint` must be the same.
+
+1. Finally the `Route` in Rahti 1:
+
+    ```yaml
+    echo 'apiVersion: route.openshift.io/v1
```

Comment:
I tried with our `course-flask-demo`. I don't know if it's the case for all the applications (maybe most of them), should we add:
```yaml
spec:
  ...
  tls:
    insecureEdgeTermination: Redirect
    termination: edge
  ...
```
?

Reply:
True, Rahti 1 has the certificate, but Rahti 2 does not.

Reply:
Does this make more sense?

    Replace `<test>` so the URL is the same as in Rahti 1. Rahti 1 will be later configured to relay the request to Rahti 2, when the request reaches Rahti 2, the Loadbalancer needs to know to which application to send the request, the host parameter is used (Using `HTTPD` headers).

Reply:
I like this change. Still a difficult concept to explain in details. This one is a good approach 👍